### PR TITLE
Removing CH, PH, HMI_MH singletones

### PR DESCRIPTION
**Pull request by Kozoriz**
Removed 3 (ConnectionHandler, ProtocolHandler, HMIMessageHandler) singletones. 
Corrected dependent tests and components.

### CODE REVIEW

**Comment by Kozoriz** [comment]
Please review : @anosach-luxoft , @AMikhniev , @AByzhynar , @LuxoftAKutsan , @VVeremjova

File: src/appMain/life_cycle.cc [hunk 1]
```diff
@@ -90,34 +90,44 @@ LifeCycle::LifeCycle()
...
+  DCHECK(transport_manager_);
+
+  DCHECK(!connection_handler_);
+  connection_handler_ = new connection_handler::ConnectionHandlerImpl(
+                          *profile::Profile::instance(),
+                          *transport_manager_);
+  DCHECK(connection_handler_);
+
+  DCHECK(!protocol_handler_);
+  //TODO(AKutsan) : APPLINK-20265 Singletons should be removed
```
Comment (AByzhynar) [inline src/appMain/life_cycle.cc]: Are you sure that TODO comment with Jira Task ID APPLINK-.... must present in open source repo?

**Comment by LuxoftAKutsan** [comment]
reviewed.

**Comment by LuxoftAKutsan** [comment]
@Kozoriz Please don't shy fix code by review notes event if same code in PASA repo. After you will finish this review I will cherri-pick commits with fixed to pasa

**Comment by AByzhynar** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by AMikhniev** [comment]
Reviewed

**Pull request merged**